vac/raw/sds.md

```diff
@@ -0,0 +1,198 @@
+- the participant MUST include the current `bloom_filter` state in the broadcast message.
+
+After broadcasting a message, the message MUST be added to the participant’s buffer of unacknowledged outgoing messages.
+
+#### Receive Message
+
+Upon receiving a message,
+- the participant MUST [review the ACK status](#review-ack-status) of messages in its unacknowledged outgoing buffer
+using the received message's causal history and bloom filter.
+- the participant MUST include the received message ID in its local bloom filter.
+- the participant MUST verify that all causal dependencies are met for the received message.
+Dependencies are met if the message IDs in the `causal_history` of the received message
+appear in the local history of the receiving participant.
+
+If all dependencies are met,
+the participant MUST [deliver the message](#deliver-message).
+If dependencies are unmet,
+the participant MUST add the message to the incoming buffer of messages with unmet causal dependencies.
+
+#### Deliver Message
+
+Triggered by the [Receive Message](#receive-message) procedure.
+
+If the received message’s Lamport timestamp is greater than the participant's local Lamport timestamp,
+the participant MUST update its local Lamport timestamp to match the received message.
+The participant MUST insert the message ID into its local log,
+based on Lamport timestamp.
+If one or more message IDs with the same Lamport timestamp already exists,
+the participant MUST follow the [Resolve Conflicts](#resolve-conflicts) procedure.
+
+#### Resolve Conflicts
+
+Triggered by the [Deliver Message](#deliver-message) procedure.
+
+The participant MUST order messages with the same Lamport timestamp in ascending order of message ID.
+
+#### Review ACK Status
+
+Triggered by the [Receive Message](#receive-message) procedure.
+
+For each message in the unacknowledged outgoing buffer,
+based on the received `bloom_filter` and `causal_history`:
+- the participant MUST mark all messages in the received `causal_history` as **acknowledged**.
+- the participant MUST mark all messages included in the `bloom_filter` as **possibly acknowledged**.
+If a message appears as **possibly acknowledged** in multiple received bloom filters,
+the participant MAY mark it as acknowledged based on probabilistic grounds,
+taking into account the bloom filter size and hash number.
+
+#### Periodic Incoming Buffer Sweep
+
+The participant MUST periodically check causal dependencies for each message in the incoming buffer.- For each message in the incoming buffer:
+- the participant MAY attempt to retrieve missing dependencies from the Store node (high-availability cache) or other peers.
+- if all dependencies of a message are met,
+the participant MUST proceed to [deliver the message](#deliver-message).
+
+If a message's causal dependencies have failed to be met after a predetermined amount of time,
+the participant MAY mark them as irretrievably lost.
```

Comment:
```suggestion
the participant MAY mark them as **irretrievably lost**.
```

Sounds like a specific state that may need consistent terminology.

Reply:
[c12ef4b](https://github.com/vacp2p/rfc-index/pull/108/commits/c12ef4b92775e5fba16b97e33dab428b1d2ef0fe)